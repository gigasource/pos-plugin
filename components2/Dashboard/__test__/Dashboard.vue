<script>
import ContentRender from "../../../components/common/ContentRender";
import {KeepAlive, onActivated, ref, h} from "vue";
import {contentView} from "./dashboard-shared";
import _ from 'lodash';


const TextField = {
  setup(props) {
    const count = ref(0);
    const init = _.debounce(() => {
      count.value ++;
    }, 10);
    init();
    onActivated(() => {
      init();
      console.log('onActivated');
    })

    return () => <p>{count.value}</p>
  }
}

const TextComp = function () {
  return <p>c</p>
}

const views = {
  TextComp,
  TextField
}

export default {
  name: "Dashboard",
  components: {
    ContentRender,
  },
  setup() {
    return () => <KeepAlive>
      {h(views[contentView.value.name])}
    </KeepAlive>
  }
}
</script>

<style scoped>

</style>
