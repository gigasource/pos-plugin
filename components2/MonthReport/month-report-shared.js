import dayjs from "dayjs";
import {formatDate} from '../utils';
import {computed, ref} from 'vue';
import cms from "cms";
import JsonFn from "json-fn";
import _ from 'lodash';

export const monthReport = ref();
export const showProductSold = ref(true)
export const showAllZNumber = ref();
export const selectedMonth = ref(dayjs().toDate());
export const selectedMonthFrom = computed(() => dayjs(selectedMonth.value).startOf('month').toDate());
export const selectedMonthTo = computed(() => dayjs(selectedMonth.value).endOf('month').toDate());
export const monthReportFrom = ref(selectedMonthFrom.value)
export const monthReportTo = ref(selectedMonthTo.value)

export function autoAssignFromTo() {
  monthReportFrom.value = selectedMonthFrom.value
  monthReportTo.value = selectedMonthTo.value
}

export const selectedPeriod = computed(() => {
  if (monthReportFrom.value && monthReportTo.value) {
    if (!start.isSame(selectedMonthFrom.value, 'day') || !end.isSame(selectedMonthTo.value, 'day')) {
      return `${formatDate(monthReportFrom.value)} - ${formatDate(monthReportTo.value)}`
    }
  }
  if (selectedMonth.value) {
    return dayjs(selectedMonth.value).format('MM.YYYY')
  }
})

//todo: test this first
export async function getMonthReport() {
  let {total, salesByCategory, salesByPayment, zNumbers, salesByCategoryName}
    = JsonFn.clone(await new Promise(
    r => cms.socket.emit('make-month-report', monthReportFrom.value, monthReportTo.value, r)));

  zNumbers = _.reduce(zNumbers, (acc, data, date) => {
    _.forEach(data, (sum, z) => {
      if (!z) return;
      const _item = {z: parseInt(z), sum: sum, date: formatDate(date)}
      acc.push(_item);
    })
    return acc
  }, [])

  monthReport.value = {total, salesByCategory, salesByPayment, zNumbers, salesByCategoryName}
}

export async function printMonthReport() {
  await new Promise(resolve => cms.socket.emit('printReport', 'MonthlyReport', {
    showAllZNumber: showAllZNumber.value,
    showProductSold: showProductSold.value,
    from: monthReportFrom.value,
    to: monthReportTo.value,
    selectedPeriod: selectedPeriod.value
  }, resolve))
}
