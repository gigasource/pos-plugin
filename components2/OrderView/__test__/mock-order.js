const mockOrder = {
  table: "10",
  _id: "5fee6600612d45b5ffdcabb7",
  items:
    [{
      groupPrinter: "Bar",
      id: "48",
      name: "Whiskey",
      price: 40,
      product: "5e784d7786e3ec0997b5e39b",
      taxes: [19, 7],
      course: 1,
      quantity: 1,
      _id: "5fee6600612d45b5ffdcabb8",
      vSum: 40,
      vTakeAway: false,
      tax: 19,
      sent: true,
      printed: true,
      vTaxSum: {
        "19": {
          tax: 6.39,
          net: 33.61,
          gross: 40
        }
      }
    }],
  cancellationItems: [],
  takeAway: false,
  status: "inProgress",
  payment: [],
  user: [],
  vSum: 40,
  vDate: "2020-12-31T17:00:00.000Z",
  vTaxSum: {
    '19': {
      tax: 6.39,
      net: 33.61,
      gross: 40
    }
  }
}

module.exports = {
  mockOrder
}
