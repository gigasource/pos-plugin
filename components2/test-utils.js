import "../jest.setup";
import _ from 'lodash';
import {config, mount} from "@vue/test-utils";
import {hooks} from "./OrderView/pos-logic";
import {disablePay, payBtnClickable, payPrintMode, showIcon} from "./OrderView/pos-logic-be";
import StubFactory from './StubFactory';

jest.mock("vue-i18n", () => {
  return {
    useI18n() {
      return {
        t(t) {
          return t; // [Consider] return `t('${t}')` to indicate that i18n has been used
        },
        locale: "en"
      };
    }
  };
});
jest.mock("vue-router", () => {
  return {
    useRoute() {
      return {
        query: {
          type: 'default'
        }
      };
    },
    useRouter() {
      return {
        push() {
        }
      };
    }
  };
});

const dayjs = require("dayjs");
export const orm = require("schemahandler");
const syncFlow = require("schemahandler/sync/sync-flow");
const syncPlugin = require("schemahandler/sync/sync-plugin-multi");
const {stringify} = require("schemahandler/utils");
//jest.useFakeTimers("modern").setSystemTime(new Date("2021-01-01").getTime());
require("mockdate").set(new Date("2021-01-01").getTime());

const delay = require("delay");

export const foodTax = {taxes: [5, 10]};
export const drinkTax = {taxes: [16, 32]};

export const cola = {name: "Cola", price: 1.3, quantity: 1, ...drinkTax};
export const fanta = {name: "Fanta", price: 2, quantity: 1, ...drinkTax};
export const rice = {name: "Rice", price: 10, quantity: 1, ...foodTax};
export const ketchup = {name: "Add Ketchup", price: 3, quantity: 1};

beforeAll(async () => {
  orm.connect({uri: "mongodb://localhost:27017"}, "myproject");
  orm.registerSchema("Order", {
    inProgress: Boolean,
    items: [{}],
    table: Number
  });
  await orm("Order").deleteMany();
  await orm("Commit").deleteMany();

  orm.plugin(syncPlugin);
  orm.plugin(syncFlow);
  orm.registerCommitBaseCollection("Order");
  orm.plugin(require("../backend/commit/orderCommit"));
  orm.registerCollectionOptions("Order");
  orm.emit("commit:flow:setMaster", true);
});

afterEach(async () => {
  await orm("Order")
    .deleteMany()
    .direct();
  await orm("Commit")
    .deleteMany()
    .direct();
});

export let actions = [];
["showOrderReceipt", "pay", "printOrder"].forEach(e =>
  hooks.on(e, () => actions.push(e))
);
export let expectArray = () => [
  "payBtnClickable",
  payBtnClickable.value,
  "payPrintMode",
  payPrintMode.value,
  "showIcon",
  showIcon.value,
  "disablePay",
  disablePay.value,
  actions
];
//</editor-fold>

//<editor-fold desc="component">
export let component, wrapper;

export function setComponent(_component) {
  component = _component;
}

/**
 *
 * @param _component
 * @param options
 * @param useDefaults indicate whether _.defaults will be use for options
 * @returns {*}
 * example: makeWrapper(Order2, {shallow: false});
 */
export const makeWrapper = (_component, options, useDefaults = false) => {
  const defaultOption = {
    props: {},
    shallow: true,
    global: {
      directives: {
        touch: true
      },
      stubs: {
        'choose-table-dialog': true,
        'dialog-change-value': true,
        'dialog-choose-popup-modifier': true,
        "dialog-config-order-item": true,
        'dialog-form-input': true,
        'dialog-multi-payment': true,
        'dialog-multi-payment2': true,
        'dialog-product-search-result': true,
        'dialog-text-filter': true,
        'g-avatar': true,
        'g-autocomplete': true,
        'g-badge': true,
        'g-btn': true,
        "g-btn-bs": true,
        'g-card': true,
        'g-card-title': true,
        'g-combobox': true,
        'g-dialog': true,
        'g-divider': true,
        "g-icon": true,
        'g-item-group': true,
        'g-menu': true,
        "g-overlay": true,
        'g-radio': true,
        'g-simple-table': true,
        'g-spacer': true,
        'g-switch': true,
        'g-table': true,
        'g-tabs': true,
        'g-tab-item': true,
        'g-text-field': true,
        'g-text-field-bs': true,
        'g-toolbar': true,
        'g-snackbar': true,
        'pos-order': true,
        'pos-quick-order-toolbar': true,
        //'pos-order-split-order': true,
        'pos-keyboard-full': true,
        'pos-order-keyboard': true,
        'pos-order-move-items': true,
        'pos-order-receipt': true,
        'pos-order-voucher-dialog': true,
        'pos-textfield-new': true,
        'g-grid-select': StubFactory('g-grid-select'),
        'pos-payment-screen-payment-methods2': true,
        'pos-payment-screen-keyboard2': true,
        'pos-restaurant-payment-order-detail2': true,
        'pos-restaurant-payment-toolbar2': true,
      },
      mocks: {
        t: a => a,
        $t: a => a,
        $filters: {
          formatCurrency(val, decimals = 2) {
            if (!val || isNaN(val) || Math.floor(val) === val) return val;
            return val.toFixed(decimals);
          }
        }
      }
    }
  }
  const optionResult = useDefaults ? _.defaults(options, defaultOption) : _.merge(defaultOption, options)
  wrapper = mount(component || _component, optionResult);
};

beforeAll(() => (config.renderStubDefaultSlot = true));
