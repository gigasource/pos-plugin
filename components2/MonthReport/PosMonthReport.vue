<script>
import {$filters} from '../AppSharedStates';
import {onActivated, ref, watch} from 'vue'
import {getMonthReport, monthReport, showProductSold} from "./month-report-shared";
import {useI18n} from "vue-i18n";
import _ from 'lodash';
import { genScopeId } from '../utils';

export default {
  name: 'PosMonthReport',
  setup() {
    const {t} = useI18n();

    getMonthReport();

    onActivated(getMonthReport);

    return genScopeId(() => (
        <div class="report-wrapper">
          <div class="report-content">
            <div class="report__sales">
              <div class="title">
                Sales
              </div>
              <g-divider dashed color="black">
              </g-divider>
              {monthReport.value && _.map(monthReport.value.salesByPayment, (amount, payment) =>
                  <div class="detail" key={`sales${payment}`}>
                    <span> {payment} </span>
                    <span>€{$filters.formatCurrency(amount)} </span>
                  </div>
              )}
              {monthReport.value && <div class="total">
                <span>
                  {t('common.total')} </span>
                <span>
                  <u>
                    € {$filters.formatCurrency(monthReport.value.total)} </u>
                </span>
              </div>}
            </div>
            <div class="report__z-number">
              <g-divider dashed color="black"/>
              <table>
                {monthReport.value && _.map(monthReport.value.zNumbers, (report, i) =>
                    <tr key={`zNumber${i}`} class="z-number">
                      <td>
                        <div class="row-flex justify-between">
                          <span>
                            {t('report.zNumber')} {report.z}: </span>
                          <span class="ml-2">
                            € {$filters.formatCurrency(report.sum)} </span>
                        </div>
                      </td>
                      <td class="pl-3">
                        {t('common.date')}: {report.date} </td>
                    </tr>
                )} </table>
              <g-divider dashed color="black"/>
            </div>
            {showProductSold.value &&
            <div class="report__product">
              <div class="title">{t('report.productSold')} </div>
              <g-divider dashed color="black"/>
              {monthReport.value && _.map(monthReport.value.salesByCategoryName, (category, categoryName) =>
                  <div key={`category${categoryName}`}>
                    <p class="category">
                      {categoryName || 'No category'} (€ {$filters.formatCurrency(monthReport.value.salesByCategory[categoryName].vSum)})
                    </p>
                    {_.map(category, ({vSum, quantity}, product) =>
                        <p class="product" key={`item${product}`}>
                          {quantity} x {product} (€ {$filters.formatCurrency(vSum)})
                        </p>
                    )}
                  </div>
              )}
            </div>}
          </div>
        </div>
    ))
  }
}
</script>

<style scoped lang="scss">
.report-wrapper {
  background: #d8d8d8;
  padding: 0 15%;
  z-index: -1;

  .report-content {
    background: white;
    padding: 32px;
    color: #1d1d26;
    overflow: auto;
    height: 100%;

    .title {
      font-weight: 700;
      font-size: 16px;
      line-height: 20px;
      margin-bottom: 12px;
      margin-top: 8px;
    }

    .report__sales {

      .detail {
        display: flex;
        justify-content: space-between;
        padding-left: 16px;
        padding-top: 8px;
        font-size: 14px;
        line-height: 25px;
        text-transform: capitalize;
      }

      .total {
        display: flex;
        justify-content: space-between;
        font-weight: 800;
        font-size: 14px;
        line-height: 30px;
        padding: 12px 0;
      }
    }

    .report__z-number {
      margin: 16px 0 32px;

      .z-number {
        font-size: 14px;
        line-height: 25px;

        + .g-divider {
          margin-top: 16px;
        }
      }

      .g-divider + .z-number {
        margin-top: 16px;
      }
    }

    .report__product {
      .category {
        font-weight: 800;
        font-size: 14px;
        line-height: 30px;
        padding-top: 8px;
      }

      .product {
        padding: 0 16px;
        font-size: 14px;
        line-height: 25px;
      }
    }
  }
}
</style>
