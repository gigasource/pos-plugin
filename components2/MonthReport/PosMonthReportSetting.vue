<script>
import { useI18n } from 'vue-i18n';
import { getMonthReport, monthReportFrom, monthReportTo, showAllZNumber, showProductSold } from './month-report-shared';
import { watch, ref, computed } from 'vue'
import { genScopeId } from '../utils';
import dayjs from 'dayjs';
import { selectedDate } from '../EndOfDay/eod-shared';

export default {
  name: 'PosMonthReportSetting',
  setup() {
    watch([() => monthReportFrom.value, () => monthReportTo.value], getMonthReport);
    const dateStringComputed = (ori) => computed({
      get() {
        return dayjs(ori.value).format('YYYY-MM-DD')
      },
      set(val) {
        ori.value = dayjs(val).toDate()
      }
    });
    const { t } = useI18n()
    return genScopeId(() =>
        <div class="setting-wrapper">
          <g-date-picker-input label={t('report.from')} v-model={dateStringComputed(monthReportFrom).value} class="mt-5"/>
          <g-date-picker-input label={t('report.to')} v-model={dateStringComputed(monthReportTo).value}/>
          <pos-switch dense label={t('report.showProductSold')} v-model={showProductSold.value}/>
          <pos-switch dense label={t('report.showZ')} v-model={showAllZNumber.value} class="mt-5"/>
        </div>)
  }
}
</script>

<style scoped lang="scss">
.setting-wrapper {
  padding: 32px;
  display: flex;
  flex-direction: column;
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.25);

  ::v-deep .g-tf-wrapper {
    .g-tf-prepend__outer {
      border: 1px solid #979797;
      border-right: none;
      border-top-left-radius: 2px;
      border-bottom-left-radius: 2px;
      background-color: #EFEFEF;
      padding: 8px;
      margin: 0;

      .g-icon {
        font-size: 20px !important;
        width: 20px !important;
        height: 20px !important;
        color: #424242 !important;
      }
    }

    fieldset {
      border: 1px solid #979797;
      border-top-right-radius: 2px;
      border-bottom-right-radius: 2px;

      .g-tf::before {
        background-color: unset;
      }

      .inputGroup {
        .input {
          div span {
            padding: 9px;
            font-size: 14px;
            line-height: 18px !important;
            color: #1D1D26;
            height: auto !important;
          }

          .g-tf-input {
            display: none;
          }
        }
      }


      .g-tf-label__active {
        transform: translate(-32px, -32px);
        color: #000;
        font-size: 14px;
        line-height: 18px;
        font-weight: 700;
      }
    }
  }

  .g-switch-wrapper {
    ::v-deep .g-switch-label {
      font-weight: 700;
      font-size: 14px;
      line-height: 18px;
      color: #000;
    }
  }
}

@media screen and (max-width: 1023px) {
  .setting-wrapper {
    padding: 8px;
    height: 100%;
    overflow: auto;

    & > div {
      margin: 16px 0 0 !important;

      ::v-deep .g-tf-wrapper {
        margin-bottom: 4px;
      }

      ::v-deep .g-switch-container {
        display: block;
      }
    }
  }
}
</style>

<style lang="scss">
@media screen and (max-height: 375px) {
  .g-picker .g-picker__title {
    display: none;
  }
}
</style>
