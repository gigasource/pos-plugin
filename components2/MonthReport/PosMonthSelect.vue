<script>
import dayjs from 'dayjs'
import {autoAssignFromTo, getMonthReport, selectedMonth, selectedPeriod} from './month-report-shared';
import {genScopeId} from '../utils';

export default {
  name: 'PosMonthSelect',
  setup() {
    const nextMonth = async function () {
      selectedMonth.value = dayjs(selectedMonth.value).add(1, 'month').toDate()
      autoAssignFromTo()
      await getMonthReport()
    }

    const previousMonth = async function () {
      selectedMonth.value = dayjs(selectedMonth.value).subtract(1, 'month').toDate()
      autoAssignFromTo()
      await getMonthReport()
    }
    getMonthReport();

    return genScopeId(() => (
        <div class="month-select-wrapper">
          <g-icon previous class="mr-4" onClick={previousMonth}>
            arrow_back_ios
          </g-icon>
          <span>
            {selectedPeriod.value} </span>
          <g-icon next class="ml-4" onClick={nextMonth}>
            arrow_forward_ios
          </g-icon>
        </div>
    ))
  }
}
</script>
<style scoped lang="scss">
.month-select-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  font-weight: 700;
  font-size: 18px;
  color: #212B35;
  background: #F0F0F0;
  box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.14);
}
</style>
