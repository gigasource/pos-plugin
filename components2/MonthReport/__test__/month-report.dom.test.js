//<editor-fold desc="declare">
import {makeWrapper, orm, wrapper} from "../../test-utils";

import {nextTick} from "vue";
import {demoData} from "../../OrderView/__test__/demoData";
import PosMonthReport from "../PosMonthReport";
import PosMonthReportSetting from "../PosMonthReportSetting";
import PosMonthSelect from "../PosMonthSelect";
import {monthReportTo} from "../month-report-shared";
import dayjs from "dayjs";
import moment from "moment";

const { prepareDb, prepareReportTest } = require('../../../backend/order-logic/report.prepare.test')

const {stringify} = require("schemahandler/utils");

const delay = require("delay")

jest.mock('cms', () => {
  process.env.USE_GLOBAL_ORM = true
  const { cmsFactory } = require('../../../test-utils')
  const _cms = cmsFactory('eodDom')
  global.cms = _cms
  return {
    socket: _cms.feSocket,
    getModel: function (modelName) {
      return _cms.orm(modelName)
    }
  }
})

let cms = global.cms

//require('../../../backend/print/print-report/report-index')(cms);

//</editor-fold>

describe("eod test", function () {
  beforeAll(async () => {
    await cms.init()
    await prepareReportTest(cms)
    await require('../../../backend/print/print-report/report-index')(cms);

    await orm("PosSetting").remove({});
    await orm("PosSetting").create(demoData.PosSetting[0]);
    await orm("OrderLayout").remove({});
    await orm("OrderLayout").create(demoData.OrderLayout[0]);
    await orm("Room").remove({});
    await orm("Room").create(demoData.Room);
    await prepareDb(/*cms.*/orm)
    cms.triggerFeConnect()
  });

  it("case 1 PosMonthReport", async function () {
    makeWrapper(PosMonthReport, {
      //shallow: true,
      props: {
      }
    });

    //await getMonthReport();
    await delay(200);
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot()
  }, 80000);

  it("case 2 PosMonthReportSetting", async function () {
    makeWrapper(PosMonthReportSetting, {
      //shallow: true,
      props: {
      }
    });

    //await getMonthReport();
    await delay(200);
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot()
  }, 80000);

  it("case 3 PosMonthSelect", async function () {
    makeWrapper(PosMonthSelect, {
      //shallow: true,
      props: {
      }
    });

    //await getMonthReport();
    wrapper.find('[previous]').trigger('click')
    await delay(200);
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot()
  }, 80000);

  it("case 4 PosMonthSelect 2", async function () {
    makeWrapper(PosMonthSelect, {
      //shallow: true,
      props: {
      }
    });
    //const a = dayjs;
    monthReportTo.value = moment('20.02.2021', 'DD.MM.YYYY').toDate();
    const a = monthReportTo
    //await getMonthReport();
    //wrapper.find('[previous]').trigger('click')
    await delay(200);
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot()
  }, 80000);

});
