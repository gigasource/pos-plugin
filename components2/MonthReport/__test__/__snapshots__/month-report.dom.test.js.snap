// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eod test case 1 PosMonthReport 1`] = `
<div class="report-wrapper">
  <div class="report-content">
    <div class="report__sales">
      <div class="title">
        Sales
      </div>
      <g-divider-stub dashed="true"
                      color="black"
      >
      </g-divider-stub>
      <div class="detail">
        <span>
          cash
        </span>
        <span>
          €17.60
        </span>
      </div>
      <div class="detail">
        <span>
          card
        </span>
        <span>
          €4.08
        </span>
      </div>
      <div class="total">
        <span>
          common.total
        </span>
        <span>
          <u>
            € 21.68
          </u>
        </span>
      </div>
    </div>
    <div class="report__z-number">
      <g-divider-stub dashed="true"
                      color="black"
      >
      </g-divider-stub>
      <table>
        <tr class="z-number">
          <td>
            <div class="row-flex justify-between">
              <span>
                report.zNumber 1:
              </span>
              <span class="ml-2">
                € 19.60
              </span>
            </div>
          </td>
          <td class="pl-3">
            common.date: 01.05.2021
          </td>
        </tr>
        <tr class="z-number">
          <td>
            <div class="row-flex justify-between">
              <span>
                report.zNumber 2:
              </span>
              <span class="ml-2">
                € 0.78
              </span>
            </div>
          </td>
          <td class="pl-3">
            common.date: 01.06.2021
          </td>
        </tr>
      </table>
      <g-divider-stub dashed="true"
                      color="black"
      >
      </g-divider-stub>
    </div>
    <div class="report__product">
      <div class="title">
        report.productSold
      </div>
      <g-divider-stub dashed="true"
                      color="black"
      >
      </g-divider-stub>
      <div>
        <p class="category">
          drink (€ 8.68)
        </p>
        <p class="product">
          4 x Cola (€ 4.68)
        </p>
        <p class="product">
          2 x Fanta (€ 4)
        </p>
      </div>
      <div>
        <p class="category">
          food (€ 13)
        </p>
        <p class="product">
          1 x Rice (€ 13)
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`eod test case 2 PosMonthReportSetting 1`] = `
<div class="setting-wrapper">
  <g-date-picker-input label="report.from"
                       modelvalue="Fri Jan 01 2021 00:00:00 GMT+0700 (Indochina Time)"
                       class="mt-5"
  >
  </g-date-picker-input>
  <g-date-picker-input label="report.to"
                       modelvalue="Sun Jan 31 2021 23:59:59 GMT+0700 (Indochina Time)"
  >
  </g-date-picker-input>
  <pos-switch dense="true"
              label="report.showProductSold"
              modelvalue="true"
  >
  </pos-switch>
  <pos-switch dense="true"
              label="report.showZ"
              class="mt-5"
  >
  </pos-switch>
</div>
`;

exports[`eod test case 3 PosMonthSelect 1`] = `
<div class="month-select-wrapper">
  <g-icon-stub previous="true"
               class="mr-4"
  >
    arrow_back_ios
  </g-icon-stub>
  <span>
    December 2020
  </span>
  <g-icon-stub next="true"
               class="ml-4"
  >
    arrow_forward_ios
  </g-icon-stub>
</div>
`;

exports[`eod test case 4 PosMonthSelect 2 1`] = `
<div class="month-select-wrapper">
  <g-icon-stub previous="true"
               class="mr-4"
  >
    arrow_back_ios
  </g-icon-stub>
  <span>
    01.01.2021 - 20.02.2021
  </span>
  <g-icon-stub next="true"
               class="ml-4"
  >
    arrow_forward_ios
  </g-icon-stub>
</div>
`;
